src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_SIZE = 10;
+    private T[] values;
+    private int size = 0;
+
+    public ArrayList() {
+        values = (T[]) new Object[DEFAULT_SIZE];
+    }
+
+    public void resizeArray() {
+        int newSize = (size + (size >> 1));
+        values = Arrays.copyOf(values, newSize);
+    }
+
+    public void checkIndex(int index) throws ArrayListIndexOutOfBoundsException {
+        if (index < 0 || index >= size) {
+            throw new ArrayListIndexOutOfBoundsException("This index doesn't exist");
```

Comment:
```suggestion
            throw new ArrayListIndexOutOfBoundsException("Index " + index + " doesn't exist in list of size " + size);
```

lets make exception messages more informative